src/transformers/generation_flax_utils.py

```diff
@@ -14,934 +14,15 @@
-            merged_is_sent_finished = jnp.concatenate([state.is_sent_finished, did_topk_just_finished], axis=1)
-            topk_merged_indices = jnp.flip(lax.top_k(merged_scores, k=num_beams)[1], axis=1)
-            next_sequences, next_scores, next_is_sent_finished = gather_beams(
-                [merged_sequences, merged_scores, merged_is_sent_finished], topk_merged_indices, batch_size, num_beams
-            )
-
-            # 8. Update model kwargs.
-            # Determine the top k beam indices from the original set of all beams.
-            # With these, gather the top k beam-associated caches.
-            next_running_indices = gather_beams(topk_beam_indices, next_topk_indices, batch_size, num_beams)
-            next_cache = gather_beams(cache, next_running_indices, batch_size, num_beams)
-            model_outputs["past_key_values"] = jax.tree_util.tree_map(lambda x: flatten_beam_dim(x), next_cache)
-            next_model_kwargs = self.update_inputs_for_generation(model_outputs, state.model_kwargs)
-
-            return BeamSearchState(
-                cur_len=state.cur_len + 1,
-                running_scores=next_running_scores,
-                running_sequences=next_running_sequences,
-                scores=next_scores,
-                sequences=next_sequences,
-                is_sent_finished=next_is_sent_finished,
-                model_kwargs=next_model_kwargs,
-            )
-
-        # The very first prompt often has sequence length > 1, so run outside of `lax.while_loop` to comply with TPU
-        if input_ids.shape[-1] > 1:
-            state = partial(beam_search_body_fn, input_ids_length=input_ids.shape[-1])(state)
-
-        if not trace:
-            state = self._run_loop_in_debug(beam_search_cond_fn, beam_search_body_fn, state)
-        else:
-            state = lax.while_loop(beam_search_cond_fn, beam_search_body_fn, state)
 
-        # Account for the edge-case where there are no finished sequences for a
-        # particular batch item. If so, return running sequences for that batch item.
-        none_finished = jnp.any(state.is_sent_finished, axis=1)
-        sequences = jnp.where(none_finished[:, None, None], state.sequences, state.running_sequences)
-        scores = jnp.where(none_finished[:, None], state.scores, state.running_scores)
+from .generation.flax_utils import FlaxGenerationMixin
 
-        # take best beam for each batch
-        sequences = sequences[:, -1]
-        scores = scores[:, -1]
 
-        return FlaxBeamSearchOutput(sequences=sequences, scores=scores)
+class FlaxGenerationMixin(FlaxGenerationMixin):
+    # warning at import time
+    warnings.warn(
+        "Importing `FlaxGenerationMixin` from `src/transformers/generation_flax_utils.py` is deprecated and will "
+        "be removed in Transformers v5. Import from `src/transformers/generation/flax_utils.py` instead.",
+        FutureWarning,
+    )
```

Comment:
Our deprecation warnings typically go in `__init__()` ([example](https://github.com/huggingface/transformers/blob/5041bc3511d098814598cf1cfc6c6bd20e72c144/src/transformers/models/wav2vec2/modeling_wav2vec2.py#L461)). However, the generation mixins have no `__init__()` -- as it stands, the warning is printed at import time.